src/apiUtils.ts

```diff
@@ -61,25 +61,16 @@ export async function fetchWithRetry(
       lastError = new Error(
         `HTTP error ${response.status}: ${response.statusText}`
       )
-
-      // Log the retry attempt
-      if (attemptCount < retry.maxRetries) {
-        const retryDelayMs =
-          retry.initialDelayMs * Math.pow(retry.backoffFactor, attemptCount)
-        core.info(
-          `Request to ${url} failed with status ${response.status}. Retrying in ${retryDelayMs}ms... (${attemptCount + 1}/${retry.maxRetries})`
-        )
-        await new Promise((resolve) => setTimeout(resolve, retryDelayMs))
-      }
     } catch (error) {
       // Network errors, timeouts, etc.
       lastError = error instanceof Error ? error : new Error(String(error))
-
+    } finally {
```

Comment:
With `finally,` this section is executed even the `try { }` block returns, so successful requests will be logged as failed attempts. There is no need to wrap it in `finally` as both `try { }` and `catch { }` will initialize the error variable if `try { }` does not return.

Reply:
After success, we are returning from the function [here](https://github.com/grafana/run-k6-action/pull/40/files#diff-15d12dfee35d8e50d55e611a57d9997992c276aa034d613ec7b24e51a50141e4R57) so the `finally` block will not be executed. If it does not return it will continue with processing the error